### PR TITLE
Fix/PoolTogether RPC optimization

### PR DESCRIPTION
**Pull request by alazarevski**
Fixes potential rate limit reached issue which happens on local builds

1. Added 15 second (avg block time) delay between triggering functions
2. Removed unnecessary RPC calls
3. Added PoolTogether thegraph calls for withdrawals and deposits
4. Added PoolTogether thegraph calls for player info and pool info
5. Added optional preselectedValue prop in ValueSelectorCard.js when joining a pool with ticket count

Todo: check it is possible to extract more info from thegraph for supplyRatePerBlock, player.ticketBalance, contract balance and accountedBalance

### CODE REVIEW

File: src/services/poolTogether.js [hunk 1]
```diff
@@ -425,16 +459,16 @@ export async function getPoolTogetherTransactions(symbol: string, address: strin
       }
       return txs;
     }, []);
-  } catch (e2) {
-    reportLog('Error getting PoolTogether withdrawal transaction logs', {
+  } catch (e) {
+    reportLog('Error getting PoolTogether transaction history', {
```
Comment (IAmKio) [inline src/services/poolTogether.js:463]: Big try / catch blocks here :( it feels like a big escape clause 😢

File: src/services/poolTogether.js [hunk 2]
```diff
@@ -117,22 +184,28 @@ export async function getPoolTogetherInfo(symbol: string, address: string): Prom
       }, Sentry.Severity.Error);
       return null;
     }
-
+    const accountedBalance = await contract.accountedBalance();
+    const balanceCallData = contract.interface.encodeFunctionData('balance');
+    const result = await provider.call({ to: contract.address, data: balanceCallData });
```
[the remark this answers is not in the document]
Comment (alazarevski) [reply src/services/poolTogether.js:189]: if any of these fail the poolprize stats cannot be obtained, exceptions will be caught in the final catch where a null result is returned after the error is logged

File: src/services/poolTogether.js [hunk 3]
```diff
@@ -84,29 +146,34 @@ export async function getPoolTogetherInfo(symbol: string, address: string): Prom
...
-        const ticketBalance = openBalance.add(userCurrentPoolBalance); // this is the current ticket balance in total
-        userInfo = {
-          openBalance: formatMoney(utils.formatUnits(openBalance.toString(), unitType)),
-          totalBalance: formatMoney(utils.formatUnits(totalBalance.toString(), unitType)),
-          userCurrentPoolBalance: formatMoney(utils.formatUnits(userCurrentPoolBalance.toString(), unitType)),
-          ticketBalance: formatMoney(utils.formatUnits(ticketBalance.toString(), unitType)),
-        };
+      // TODO user players[0].latestBalance / consolidatedBalance when it gets fixed and deployed on thegraph.com
+      if (!isEmpty(players)) {
+        const totalBalance = await contract.totalBalanceOf(address); // open, committed, fees, sponsorship balance
```
[the remark this answers is not in the document]
Comment (alazarevski) [reply src/services/poolTogether.js:170]: userInfo can be null, except a null will be returned and break execution in the case of an error

File: environments/.env.staging [hunk 4]
```diff
@@ -42,3 +42,4 @@ POOL_DAI_CONTRACT_ADDRESS=0xC3a62C8Af55c59642071bC171Ebd05Eb2479B663
 POOL_USDC_CONTRACT_ADDRESS=0xa0B2A98d0B769886ec06562ee9bB3572Fa4f3aAb
 DAI_ADDRESS=0x4F96Fe3b7A6Cf9725f59d353F723c1bDb64CA6Aa
 USDC_ADDRESS=0x75B0622Cec14130172EaE9Cf166B92E5C112FaFF
+POOLTOGETHER_GRAPH_ID=QmSmtdcYD9Zuk7W9zuk73rRCNema26vMdF7muyfGwLzqfT
```
[the remark this answers is not in the document]
Comment (alazarevski) [reply environments/.env.staging:45]: Yes. Unfortunately PoolTogether will work only with prod values for now

File: src/services/poolTogether.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (IAmKio) [reply src/services/poolTogether.js]: Thanks for the explanation.

File: src/services/poolTogether.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (IAmKio) [reply src/services/poolTogether.js]: Cheers mate.

File: src/services/poolTogether.js [hunk 5]
```diff
@@ -27,6 +27,7 @@ import {
   DAI_ADDRESS,
   USDC_ADDRESS,
   POOLTOGETHER_GRAPH_ID,
+  THE_GRAPH_API_URL,
```
Comment (IAmKio) [inline src/services/poolTogether.js:30]: 👍

**Comment by IAmKio** [comment]
Thanks for the updates! Let's get this into 2.12.1!

**Pull request merged**